src/Member.ts

```diff
@@ -51,25 +57,43 @@ export class Member {
       const answerSdp = await this.rtcpPeer.createAnswer(options);
       await this.rtcpPeer.setLocalDescription(answerSdp);
       // Send the answer to the remote peer through the signaling server.
-      await this.plugin.sendMessage({
-        request: 'start',
-        room: this.plugin.room_id
-      }, answerSdp, {handle_id: this.handleId});
+      await retryPromise(
+        () => this.plugin.sendMessage({
+          request: 'start',
+          room: this.plugin.room_id
+        }, answerSdp, {handle_id: this.handleId})
+      ).catch(() => {
+        this.plugin?.session.emit('disconnected');
+        this.plugin.session.offAll()
+      });
 
       this.stream = event.stream;
 
-      this.plugin?.session.emit('member:join', this.memberInfo)
+      this.plugin?.session.emit('member:update', this.memberInfo)
     }
 
     // Send ICE events to Janus.
     const RTCPeerOnIceCandidate = (event) => {
-      if (this.rtcpPeer.signalingState !== 'stable') return;
-      this.plugin.sendTrickle(event.candidate || null);
+      if (event.candidate) {
+        this.plugin.sendTrickle(event.candidate);
+      }
     }
 
-    this.rtcpPeer = new RTCPeerConnection();
+    const RTCPeerOnConnectionStateChange = async () => {
+      if (this.rtcpPeer.iceConnectionState === 'disconnected') {
+        this.plugin.session.emit('member:disconnected', this.memberInfo)
+        delete this.plugin.memberList[this.memberInfo.id]
+        await this.plugin.syncParticipants()
```

Comment:
Let's use `try/catch` instead. It's a bit cleaner and easier to follow
```
try {
  await this.plugin.syncParticipants()
} catch (err) {
  this.plugin.session.emit('disconnected')
  this.plugin.session.offAll()
}
```

Reply:
done